lectures/MobilePlatforms/ReadMe.md

```diff
@@ -0,0 +1,176 @@
+
+* Differences between desktops and mobile devices w.r.t. applications, security, energy, and other related aspects
+* Challenges and opportunities created by mobile platforms
+* Brief specifics of the Android stack and how applications are structured
+* A few ideas for offering users a good experience on their mobile
+* The ecosystem that mobile apps plug into
+
+## From desktops to mobiles
+
+Roughly every decade, a new, lower priced computer class forms, based on a new programming platform, network, and interface. This results in new types of usage, and often the establishment of a new industry. This is known as [Bell's Law](https://en.wikipedia.org/wiki/Bell%27s_law_of_computer_classes).
+
+With every new computer class, the number of computers per person increases drastically. Today we have clouds of vast data centers, and perhaps an individual computer, like our laptop, that we use to be productive. On top of that come several computer devices per individual, like phones, wearables, and smart home items, which we use for entertainment, communication, quality of life, and so on.
+
+It is in this context that mobile software development becomes super-important.
+
+We said earlier that, no matter what job you will have, you will write code. We can add to that: you will likely write code for mobile devices. There are more than 15 billion mobile devices operating worldwide, and that number is only going up. As Gordon Bell said, this leads to new usage patterns.
+
+We access the Internet more often from our mobile than our desktop or laptop. Most of the digital content we consume we do so on mobiles. We spend hours a day on our mobile, and the vast majority of that time we spend in apps, not on websites.
+
+A simple example of a major change in how we use computing and communication is social media. Most of the world's population uses it. It changes how we work. Even the professional workforce is increasingly dependent on mobiles, for this reason.
+
+### Mobile vs. desktop: Applications
+
+There are many differences between how we write applications for a mobile device vs. a desktop computer. On a desktop, applications can do pretty much whatever they want, whereas, on a mobile, each app is super-specialized. On the desktop, users explicitly starts applications; on mobile, the difference between running or not is fluid: apps can be killed anytime, and they need to be ready to restart.
+
+On a desktop, you typically have multiple applications active in the foreground, with multiple windows on-screen. The mobile experience is difference: a user's interaction with an app doesn't always begin in the same place, but rather the user's journey often begins non-deterministically. As a result, a mobile app has a more complex structure than a traditional desktop application, with multiple types of components and entry points.
```

Comment:
I'm not sure what "non-deterministically" means here, the user doesn't tap buttons at random?

Reply:
Meaning it doesn't start in the same way. The composition of an email could start from the email app, from a photo share, etc.

Reply:
I would say something like "the user's journey can start from different places" then, since it's still deterministic as far as the user is concerned.